spec/factories/lettings_log.rb

```diff
@@ -168,14 +159,15 @@
       town_or_city { "London" }
       ppcodenk { 1 }
       tshortfall_known { 1 }
-    end
-    trait :completed2024 do
-      completed
-      address_line1_input { address_line1 }
-      postcode_full_input { postcode_full }
-      nationality_all_group { 826 }
-      uprn { 1 }
-      uprn_selection { 1 }
+      after(:build) do |log, _evaluator|
+        if log.startdate >= Time.zone.local(2024, 4, 1)
+          log.address_line1_input = log.address_line1
+          log.postcode_full_input = log.postcode_full
+          log.nationality_all_group = 826
+          log.uprn = "10033558653"
+          log.uprn_selection = 1
+        end
+      end
```

Comment:
this is within the :completed trait and makes 24/25 logs with the completed trait actually be completed.

Reply:
Nice

Reply:
Are there any pre-2024 only fields we'd want this to clear?

Reply:
I'm not aware, but the test I have put this to is that if I create logs for before or after, then after being saved, validated, derived, cleared, etc, then they do come back with the correct status